pcdsdevices/sqr1.py

```diff
@@ -0,0 +1,142 @@
+"""
+    Ophyd class for square-one tri-sphere motion system
+    (https://www.sqr-1.com/tsp.html)
+"""
+
+import logging
+import typing
+
+from ophyd.device import Component as Cpt
+from ophyd.device import Device
+from ophyd.device import FormattedComponent as FCpt
+from ophyd.pv_positioner import PVPositionerIsClose
+from ophyd.signal import EpicsSignal
+from ophyd.status import wait as status_wait
+
+logger = logging.getLogger(__name__)
+
+
+class SQR1Axis(PVPositionerIsClose):
+    setpoint = FCpt(EpicsSignal, '{prefix}:TARGET:{_axis}', kind='normal')
+    readback = FCpt(EpicsSignal, '{prefix}:TARGET:{_axis}:RBV', kind='hinted')
+    actuate = Cpt(EpicsSignal, '{prefix}:MOV', kind='normal')
+    actuate_value = 1
+    stop_signal = Cpt(EpicsSignal, '{prefix}:KILL', kind='normal')
+    stop_value = 1
+
+    def __init__(self,
+                 prefix,
+                 axis: str,
+                 sync_setpoints: typing.Callable | None = None,
+                 **kwargs
+                 ):
+        self._axis = axis  # axis values {X, Y, Z, rX, rY, rZ}
+        self._sync_setpoints = sync_setpoints
+        super().__init__(prefix, **kwargs)
+
+    def move(self,
+             position: float,
+             wait: bool = True,
+             timeout: float = None,
+             moved_cb: float = None,
+             sync_enable: bool = True,
+             ):
+        if sync_enable and self._sync_setpoints:
+            self._sync_setpoints()
+        return super().move(position, wait, timeout, moved_cb)
+
+
+class SQR1(Device):
+    x = Cpt(SQR1Axis, '{prefix}', axis="X", rtol=0.001)
```

Comment:
These are all malformed now:
```
In [1]: import pcdsdevices.sqr1

In [2]: dev = pcdsdevices.sqr1.SQR1("MCQ:TMO:01", name="dev")

In [3]: dev.component_names
Out[3]: ('x', 'y', 'z', 'rx', 'ry', 'rz', 'stop_signal')

In [4]: dev.x.prefix
Out[4]: 'MCQ:TMO:01{prefix}'
```
the curly braces setup only works for FCpt, to refactor to Cpt you would have needed to drop this down to (I think):
```suggestion
    x = Cpt(SQR1Axis, '', axis="X", rtol=0.001)
```